packages/react-components/src/organisms/Form.tsx

```diff
@@ -53,11 +51,12 @@ const Form = <T extends void | Record<string, unknown>>({
     if (formRef.current!.reportValidity() && parentValidation) {
       setStatus('isSaving');
       try {
-        const result = await onSave();
-        if (formRef.current) {
-          setStatus('hasSaved');
-        }
-        return result;
+        return await onSave().then((result) => {
```

Comment:
I think we can remove this promise chain to improve readability

Reply:
Done